From 1cec05857c5aaa0eebe75e7c2e6c6116fb9759cd Mon Sep 17 00:00:00 2001
From: Marcelo Tosatti <mtosatti@redhat.com>
Date: Mon, 23 Sep 2013 20:00:13 +0200
Subject: [PATCH 29/29] fix steal time MSR vmsd callback to proper opaque type

RH-Author: Marcelo Tosatti <mtosatti@redhat.com>
Message-id: <20130923200013.GA20765@amt.cnet>
Patchwork-id: 54498
O-Subject: [RHEL7 qemu-kvm PATCH] fix steal time MSR vmsd callback to proper opaque type
Bugzilla: 903889
RH-Acked-by: Andrew Jones <drjones@redhat.com>
RH-Acked-by: Gleb Natapov <gleb@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>

BZ: 903889

commit 0e5035776df31380a44a1a851850d110b551ecb6 of uq/master branch of qemu-kvm.git repo

Convert steal time MSR vmsd callback pointer to proper X86CPU type.

Signed-off-by: Marcelo Tosatti <mtosatti@redhat.com>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 target-i386/machine.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/target-i386/machine.c b/target-i386/machine.c
index 3659db9..ecf1617 100644
--- a/target-i386/machine.c
+++ b/target-i386/machine.c
@@ -294,9 +294,9 @@ static bool pv_eoi_msr_needed(void *opaque)
 
 static bool steal_time_msr_needed(void *opaque)
 {
-    CPUX86State *cpu = opaque;
+    X86CPU *cpu = opaque;
 
-    return cpu->steal_time_msr != 0;
+    return cpu->env.steal_time_msr != 0;
 }
 
 static const VMStateDescription vmstate_steal_time_msr = {
@@ -305,7 +305,7 @@ static const VMStateDescription vmstate_steal_time_msr = {
     .minimum_version_id = 1,
     .minimum_version_id_old = 1,
     .fields      = (VMStateField []) {
-        VMSTATE_UINT64(steal_time_msr, CPUX86State),
+        VMSTATE_UINT64(env.steal_time_msr, X86CPU),
         VMSTATE_END_OF_LIST()
     }
 };
-- 
1.7.1

